common/elasticsearch/aws.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package elasticsearch
+
+import (
+	"fmt"
+	awsauth "github.com/smartystreets/go-aws-auth"
+	"net/http"
+	"os"
+)
+
+type AWSSigningTransport struct {
+	HTTPClient  *http.Client
+	Credentials awsauth.Credentials
+}
+
+// RoundTrip implementation
+func (a AWSSigningTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return a.HTTPClient.Do(awsauth.Sign4(req, a.Credentials))
+}
+
+func createAWSClient() (*http.Client, error) {
+	id := os.Getenv("AWS_ACCESS_KEY_ID")
```

Comment:
How to deploy heapster, kubernetes or standalone?  
And if using kubernetes to deploy, how to inject this environment variable? 
Would you like to explain me the details of deployment?
Thanks so much.